pkg/scheduler/framework/plugins/volumebinding/volume_binding.go

```diff
@@ -470,20 +472,41 @@ func (pl *VolumeBinding) Score(ctx context.Context, cs *framework.CycleState, po
 	if !ok {
 		return 0, nil
 	}
-	// group by storage class
+
 	classResources := make(classResourceMap)
-	for _, staticBinding := range podVolumes.StaticBindings {
-		class := staticBinding.StorageClassName()
-		storageResource := staticBinding.StorageResource()
-		if _, ok := classResources[class]; !ok {
-			classResources[class] = &StorageResource{
-				Requested: 0,
-				Capacity:  0,
+	if len(podVolumes.StaticBindings) != 0 || !pl.fts.EnableStorageCapacityScoring {
+		// group static binding volumes by storage class
+		for _, staticBinding := range podVolumes.StaticBindings {
+			class := staticBinding.StorageClassName()
+			storageResource := staticBinding.StorageResource()
+			if _, ok := classResources[class]; !ok {
+				classResources[class] = &StorageResource{
+					Requested: 0,
+					Capacity:  0,
+				}
+			}
+			classResources[class].Requested += storageResource.Requested
+			classResources[class].Capacity += storageResource.Capacity
+		}
+	} else {
+		// group dynamic binding volumes by storage class
+		for _, provision := range podVolumes.DynamicProvisions {
+			if provision.Capacity == nil {
+				continue
+			}
+			class := *provision.PVC.Spec.StorageClassName
+			if _, ok := classResources[class]; !ok {
+				classResources[class] = &StorageResource{
+					Requested: 0,
+					Capacity:  0,
+				}
 			}
+			requestedQty := provision.PVC.Spec.Resources.Requests[v1.ResourceName(v1.ResourceStorage)]
+			classResources[class].Requested += requestedQty.Value()
+			classResources[class].Capacity += provision.Capacity.Capacity.Value()
```

Comment:
Why is the scheduler *summing* node capacities? If I have a Pod that uses 2 50GiB PVCs of the same StorageClass, and a node with 100GiB capacity for that class, I would expect the node to get full (100%). With added the node capacity for each PVC here (???), the node gets 50% full. That seems very odd. Please add an unit test for this case.

(Question is what capacity to use if each `DynamicProvisions` reports a different capacity for the same storage class... Use `max()` or `min()`?

In static provisioning, the PV capacity is considered, not the node capacity. So summing them makes some sense.

Reply:
I have completed addressing this matter.

> Why is the scheduler summing node capacities?

This was a mistake in the implementation. I have fixed it.

> Please add an unit test for this case.

[I have added it.](https://github.com/kubernetes/kubernetes/pull/128184/files#diff-bf8df5283cbffef0aed1bc2d5b1a0575021e4cf0ce27bd1d70b48dfec1343b1dR866-R901)

> (Question is what capacity to use if each `DynamicProvisions` reports a different capacity for the same storage class... Use `max()` or `min()` or what?

In that case, I think we should use the most recent value, so [I have corrected it accordingly.](https://github.com/kubernetes/kubernetes/pull/128184/files#diff-75e4458878912a7aef522a2eebd4928b91d7622445f2da897119450016cc8a49R504)